### PR TITLE
Add resolvers and mutations for TransactionAssets

### PR DESCRIPTION
**Pull request by richthegeek**
This adds the `assets` resolver to the `transaction` response.

It also adds the mutation helpers:
* `createTransactionAsset({ transactionId, name, filetype })`
* `finaliseTransactionAssetUpload({ assetId })`
* `deleteTransactionAsset({ assetId })`

### CODE REVIEW

File: tests/graphql/transaction.spec.ts [hunk 1]
```diff
@@ -436,6 +440,122 @@ describe("Transaction", () => {
...
+        filetype: "jpg",
+        thumbnail: "...",
+        fullsize: "...",
+      };
+
+      stub.resolves({
+        finaliseTransactionAssetUpload: asset,
+      } as any);
+
+      // act: update transaction with prepared split data
```
Comment (dmytroyarmak) [inline tests/graphql/transaction.spec.ts:514]: 🙃 it looks like this comment is not relevant

File: tests/graphql/transaction.spec.ts [hunk 2]
```diff
@@ -436,6 +440,122 @@ describe("Transaction", () => {
...
+
+    it("should call rawQuery once using proper arguments and return a MutationResult", async () => {
+      const assetId = "000000000000-0000-0000-0000-00000000";
+
+      stub.resolves({
+        deleteTransactionAsset: { success: true },
+      } as any);
+
+      // act: update transaction with prepared split data
+      const result = await client.models.transaction.deleteTransactionAsset({ assetId });
```
Comment (dmytroyarmak) [inline tests/graphql/transaction.spec.ts:549]: 🙃 comment is incorrect

File: lib/graphql/schema.ts [hunk 3]
```diff
@@ -527,6 +527,12 @@ export type Mutation = {
   updateTransactionSplits: Transaction;
   /** Delete transaction splits */
   deleteTransactionSplits: Transaction;
+  /** Create transaction asset */
+  createTransactionAsset: CreateTransactionAssetMutationResult;
+  /** Create transaction asset */
+  finaliseTransactionAssetUpload: TransactionAsset;
+  /** Create transaction asset */
+  deleteTransactionAsset: MutationResult;
```
Comment (johannespfeiffer) [inline lib/graphql/schema.ts:535]: These comments should probably not always be the same. Problem comes from backend.

Comment (richthegeek) [reply lib/graphql/schema.ts:535]: :+1: good catch, will update now.

File: lib/graphql/transaction.ts [hunk 4]
```diff
@@ -159,6 +173,46 @@ export const UPDATE_SPLIT_TRANSACTION = `mutation updateTransactionSplits(
...
+    assetId
+    url
+    formData {
+      key
+      value
+    }
+  }
+}`;
+
+export const FINALISE_TRANSACTION_ASSET = `mutation finaliseTransactionAssetUpload(
```
Comment (johannespfeiffer) [inline lib/graphql/transaction.ts:195]: In general, we try to stick with American English variant, so please use `finalize` (US) instead of `finalise` (UK). Added a section in Notion to make this clear for the future.

Comment (richthegeek) [reply lib/graphql/transaction.ts:195]: I see - is this a blocker? Only because it requires a backend PR to alter it there also.

File: lib/graphql/schema.flow.js [hunk 5]
```diff
@@ -1764,6 +1819,7 @@ export type UserMetadata = {|
   marketingConsentAccepted: $ElementType<Scalars, 'Boolean'>,
   phoneNumberVerificationRequired: $ElementType<Scalars, 'Boolean'>,
   signupCompleted: $ElementType<Scalars, 'Boolean'>,
+  categorizationScreenShown?: ?$ElementType<Scalars, 'Boolean'>,
```
Comment (goodfellamilos) [inline lib/graphql/schema.flow.js:1822]: 🙃 Not part of this PR, should be updated here: https://github.com/kontist/js-sdk/pull/111. Will ask @p-janik if it can be merged.

Comment (goodfellamilos) [reply lib/graphql/schema.flow.js:1822]: Update: Merged, can you please update your branch]?

Comment (richthegeek) [reply lib/graphql/schema.flow.js:1822]: This was added automatically when running `schema:generate` - I guess I could cherry-pick only the Asset-related parts?

Comment (goodfellamilos) [reply lib/graphql/schema.flow.js:1822]: Yes, I know, but now you can just update your branch with develop, it was already added there with #111.

Comment (richthegeek) [reply lib/graphql/schema.flow.js]: Done.
